### ISSUE TITLE
Datatable: (accessibility) narrator doesn't announce sortable column

### ISSUE THREAD

**Issue by arnodemer** [primefaces/primereact#6056]
### Describe the bug

Screen reader (NVDA) on datatable column header is able to read sorted column (including sort order),
but, when a column is not yet sorted, screen reader did not said anything, then user cannot guess it's a sortable column.

### PrimeReact version

10.5.1

### React version

17.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

### Steps to reproduce the behavior

1. go to data table sort showcase:  https://primereact.org/datatable/#sort
2. make a column sorted, ie: "Category"
2. start screen reader, set focus on various column header
3. if column is not sorted, reader did not announce the focused column is sortable (occurs for "Name" column in screenshot below)
4. if column is sorted, reader read correctly, sort and sort direction.  (in green in screenshot)
![image](https://github.com/primefaces/primereact/assets/153746860/46d107ac-5b2a-4f5d-b3f3-fa9f23889d9b)

### Expected behavior

Expect that reader tell to (blind) people that the column header is sortable, otherwise they cannot guess they can sort the column.

**Comment by melloware** [comment]
I am not sure Datatable has gotten all ARIA accessibility updates yet. Similar to: https://github.com/primefaces/primereact/issues/5594
